src/tools/pbs_python.c

```diff
@@ -2310,11 +2310,37 @@ main(int argc, char *argv[], char *envp[])
 		char *largv[3];
 		int ll;
 		char *pc, *pc2;
+
+#ifdef SYSTEM_PYTHON_PATH
+		snprintf(python_path, MAXPATHLEN, "%s", SYSTEM_PYTHON_PATH);
+		pc = strdup(SYSTEM_PYTHON_PATH);
+		if(pc == NULL) {
+			fprintf(stderr, "Out of memory\n");
+			return 1;
+		}
+		pc2 = strstr(pc,"bin/python");
+		if(pc2 == NULL) {
+			fprintf(stderr, "Python executable not found!\n");
+			return 1;
+		}
+		ll = pc2 - pc;
+		if(ll <= 0) {
+			fprintf(stderr, "Python home not found!\n");
+			return 1;
+		}
+		snprintf(python_prefix, ll, "%s", SYSTEM_PYTHON_PATH);
```

Comment:
I think I may have not have have been clear in what I was suggesting.  You use pc and pc2 only to calculate ll.  You then use ll as the second argument to snprintf().  The way you are only copying the prefix is by the use of ll.  I was suggesting you do a *pc = '\0'.  This way you could copy the string directly.  The usual use of the second argument of snprintf() is to prevent buffer overruns.  Since this is used differently here, it took me a little bit to realize what was going on.  What are your opinions on this?  If you feel strongly on your use of snprintf() as is, then it's fine.